packages/remix/src/client/performance.tsx

```diff
@@ -65,28 +62,31 @@ export function remixRouterInstrumentation(useEffect: UseEffect, useLocation: Us
     startTransactionOnPageLoad = true,
     startTransactionOnLocationChange = true,
   ): void => {
-    const initPathName = getInitPathName();
-    if (startTransactionOnPageLoad && initPathName) {
-      activeTransaction = customStartTransaction({
-        name: initPathName,
-        op: 'pageload',
-        origin: 'auto.pageload.remix',
-        tags: DEFAULT_TAGS,
-        metadata: {
-          source: 'url',
-        },
-      });
-    }
-
     _useEffect = useEffect;
     _useLocation = useLocation;
     _useMatches = useMatches;
 
     _customStartTransaction = customStartTransaction;
+    _startTransactionOnPageLoad = startTransactionOnPageLoad;
     _startTransactionOnLocationChange = startTransactionOnLocationChange;
   };
 }
 
+const getTracingContextFromRouteMatches = (
+  matches: RouteMatch<string>[],
+): {
+  traceparentData?: TraceparentData;
+  dynamicSamplingContext?: Record<string, unknown>;
+  propagationContext: PropagationContext;
+  id: string;
+} => {
+  const currentRouteMatch = matches[matches.length - 1];
+
+  const { sentryTrace, sentryBaggage } = currentRouteMatch.data || {};
+
+  return { id: currentRouteMatch.id, ...tracingContextFromHeaders(sentryTrace, sentryBaggage) };
```

Comment:
This makes an unnecessary copy. I recommend something like this:

```suggestion
  const out = tracingContextFromHeaders(sentryTrace, sentryBaggage);
  out.id = currentRouteMatch.id;
  return out;
```